src/DetailsView/components/requirement-view-next-requirement-configuration.tsx

```diff
@@ -39,33 +47,48 @@ export const getNextRequirementConfigurationForAssessment = (
         props.currentAssessment.requirements[requirementIndex + 1] ?? null;
     const nextRequirementVisualizationType: VisualizationType =
         props.assessmentNavState.selectedTestType;
-    return {
-        nextRequirement,
-        nextRequirementVisualizationType,
-    };
+    return (
+        <NextRequirementButton
+            deps={props.deps}
+            nextRequirement={nextRequirement}
+            nextRequirementVisualizationType={nextRequirementVisualizationType}
+            className={props.className}
+        />
+    );
 };
 
 export const getNextRequirementConfigurationForQuickAssess = (
     props: GetNextRequirementConfigurationProps,
-) => {
+): JSX.Element => {
     if (props.currentAssessment.key === AutomatedChecks.key) {
         return getNextRequirementConfigurationForAssessment(props);
     }
+
     const requirementIndex = props.deps.mediumPassRequirementKeys.findIndex(
         r => r === props.currentRequirement.key,
     );
     if (requirementIndex === props.deps.mediumPassRequirementKeys.length - 1) {
-        //TODO: special go to assessment button
-        return {
-            nextRequirement: null,
-            nextRequirementVisualizationType: props.assessmentNavState.selectedTestType,
-        };
-    } else {
-        const nextRequirementKey = props.deps.mediumPassRequirementKeys[requirementIndex + 1];
-        const nextAssessment = props.deps.getProvider().forRequirementKey(nextRequirementKey);
-        return {
-            nextRequirement: nextAssessment!.requirements.find(r => r.key === nextRequirementKey),
-            nextRequirementVisualizationType: nextAssessment!.visualizationType,
-        };
+        return (
+            <PrimaryButton
+                text={'Complete'}
+                className={props.className}
+                onClick={
+                    props.deps.dataTransferViewController.showQuickAssessToAssessmentConfirmDialog
+                }
+            ></PrimaryButton>
+        );
     }
+
+    const nextRequirementKey = props.deps.mediumPassRequirementKeys[requirementIndex + 1];
+    const nextAssessment = props.deps.getProvider().forRequirementKey(nextRequirementKey);
+    return (
+        <NextRequirementButton
+            deps={props.deps}
+            nextRequirement={
+                nextAssessment!.requirements.find(r => r.key === nextRequirementKey) as Requirement
```

Comment:
This cast was required because adding to strict null checks exception list for some reason is not working.


Failure here: https://github.com/microsoft/accessibility-insights-web/actions/runs/3916223638/jobs/6695140801

Reply:
If you want to avoid the cast, you can pull `nextRequirement` out into a variable and then passing `nextRequirement!` into the component. We know this requirement will exist because we found the Assessment based on requirement key to begin with.